src/BuildingRegistry.Api.BackOffice.Handlers.Lambda/Handlers/BuildingUnit/AttachAddressToBuildingUnitLambdaHandler.cs

```diff
@@ -0,0 +1,90 @@
+namespace BuildingRegistry.Api.BackOffice.Handlers.Lambda.Handlers.BuildingUnit
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Abstractions.Validation;
+    using Be.Vlaanderen.Basisregisters.AggregateSource;
+    using Be.Vlaanderen.Basisregisters.Api.Exceptions;
+    using Be.Vlaanderen.Basisregisters.Sqs.Exceptions;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Handlers;
+    using Be.Vlaanderen.Basisregisters.Sqs.Lambda.Infrastructure;
+    using Be.Vlaanderen.Basisregisters.Sqs.Responses;
+    using BuildingRegistry.Api.BackOffice.Abstractions;
+    using BuildingRegistry.Api.BackOffice.Abstractions.Building.Validators;
+    using BuildingRegistry.Building;
+    using BuildingRegistry.Building.Exceptions;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Configuration;
+    using Requests.BuildingUnit;
+    using TicketingService.Abstractions;
+
+    public sealed class AttachAddressToBuildingUnitLambdaHandler : BuildingUnitLambdaHandler<AttachAddressToBuildingUnitLambdaRequest>
+    {
+        private readonly BackOfficeContext _backOfficeContext;
+
+        public AttachAddressToBuildingUnitLambdaHandler(
+            IConfiguration configuration,
+            ICustomRetryPolicy retryPolicy,
+            ITicketing ticketing,
+            IIdempotentCommandHandler idempotentCommandHandler,
+            IBuildings buildings,
+            BackOfficeContext backOfficeContext)
+            : base(
+                configuration,
+                retryPolicy,
+                ticketing,
+                idempotentCommandHandler,
+                buildings)
+        {
+            _backOfficeContext = backOfficeContext;
+        }
+
+        protected override async Task<ETagResponse> InnerHandle(AttachAddressToBuildingUnitLambdaRequest request, CancellationToken cancellationToken)
+        {
+            var cmd = request.ToCommand();
+
+            try
+            {
+                await IdempotentCommandHandler.Dispatch(
+                    cmd.CreateCommandId(),
+                    cmd,
+                    request.Metadata,
+                    cancellationToken);
+            }
+            catch (IdempotencyException)
+            {
+                // Idempotent: Do Nothing return last etag
+            }
+
+            _backOfficeContext.BuildingUnitAddressRelation.Add(new BuildingUnitAddressRelation(cmd.BuildingPersistentLocalId, cmd.BuildingUnitPersistentLocalId, cmd.AddressPersistentLocalId));
```

Comment:
Shouldn't we **only** be doing this when **no** `IdempotencyException` is thrown? @ArneD what do you think?

Reply:
if this is a problem, then this should also be changed in other registers

Reply:
true, our resilience projection should fix relation is it happend to crash anyway.
Then yes, it can be done in other registries where relations are saved.